packages/ui/v2/core/WizardForm.tsx

```diff
@@ -63,10 +68,7 @@ function WizardForm<T extends DefaultStep>(props: {
                   type="submit"
                   color="primary"
                   form={`wizard-step-${step}`}
-                  className="relative ml-2"
-                  onClick={() => {
-                    setStep(step + 1);
-                  }}>
```

Comment:
This is the fundamental part of this PR, fixes validation by not switching to the next step on click, not letting the UI show errors to let the use correct them.